src/audio/AudioManager.cpp

```diff
@@ -0,0 +1,114 @@
+#include <Logger.h>
+#include <audio/AudioException.h>
+#include <audio/AudioManager.h>
+#include <ecs/View.h>
+
+
+namespace arch::audio {
+
+AudioManager::AudioManager(SoundBank* soundBank, ecs::Domain* domain, std::mutex& ecsMutex):
+	_soundBank(soundBank),
+	_domain(domain),
+	_ecsMutex(ecsMutex){
+	for(int i=0; i<16; i++) {
+		_sources[i].initialize(soundBank);
+	}
+	Logger::info("Audio system: opened audio manager");
+}
+
+AudioManager::~AudioManager() {
+	Logger::info("Audio system: closed audio manager");
+}
+
+int AudioManager::_findEmptyPlayer() {
+	for(int i=0; i<16; i++) {
+		if(!_sourceUsed[i]) {
+			return i;
+		}
+	}
+	return -1;
+}
+
+void AudioManager::play() {
+	Logger::info("Audio system: audio manager started playing");
+	while (_isListening) {
+		_updateListener();
+		{
+			auto lock = std::lock_guard(_ecsMutex);
+			for (auto&& [source] : _domain->view<AudioSource>().components()) {
```

Comment:
Audio system update, should iterate over const Components, as they should not contain state, only settings.